code/counter-app/src/main.rs

```diff
@@ -0,0 +1,175 @@
+// ANCHOR: imports
+use crossterm::event::{self, Event, KeyCode, KeyEvent, KeyEventKind};
+use ratatui::{prelude::*, widgets::*};
+// ANCHOR_END: imports
+
+// ANCHOR: modules
+mod tui;
+// ANCHOR_END: modules
+
+// ANCHOR: main
+fn main() -> color_eyre::Result<()> {
+    color_eyre::install()?;
+    tui::install_panic_hook();
+    let mut terminal = tui::init()?;
+    let app_result = App::default().run(&mut terminal);
+    tui::restore()?;
+    app_result
+}
+// ANCHOR_END: main
+
+// ANCHOR: app
+#[derive(Debug, Default)]
+pub struct App {
+    counter: u8,
+    app_state: AppState,
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+enum AppState {
+    #[default]
+    Running,
+    Done,
+}
+// ANCHOR_END: app
+
+// ANCHOR: impl App
+impl App {
+    // ANCHOR: run
+    pub fn run(
+        &mut self,
+        terminal: &mut Terminal<impl Backend>,
+    ) -> color_eyre::Result<()> {
+        while self.app_state != AppState::Done {
+            terminal.draw(|frame| self.render_frame(frame))?;
+            self.update()?;
+        }
+        Ok(())
+    }
+    // ANCHOR_END: run
+
+    // ANCHOR: render_frame
+    fn render_frame(&self, frame: &mut Frame) {
+        let counter_text = format!("Counter: {}", self.counter);
+        let instruction_text =
+            "Decrement: 'j'/Down Increment:'k'/Up Quit: 'q'/Esc";
```

Comment:
Super minor nit: Maybe we should use `Left` and `Right` instead? A number line is typically represented horizontally.

I was looking at SvelteKit's counter demo app recently and it looks like this:

<img width="310" alt="image" src="https://github.com/ratatui-org/website/assets/1813121/761592e0-a1bc-4f98-927f-abcb50a96110">

And I think let makes more intuitive sense that decrement is on the left and increment is on the right; versus whether decrement/increment is on the bottom/top or top/bottom.

Reply:
We can remove `j` and `k` and only use `Left` and `Right`?

Reply:
Yep :)

Reply:
Done